pkg/gadgets/trace/tcp/tracer/tracer.go

```diff
@@ -222,41 +190,37 @@ func (t *Tracer) run() {
 			continue
 		}
 
-		eventC := (*C.struct_event)(unsafe.Pointer(&record.RawSample[0]))
+		eventC := (*tcptracerEvent)(unsafe.Pointer(&record.RawSample[0]))
 
 		event := types.Event{
 			Event: eventtypes.Event{
 				Type: eventtypes.NORMAL,
 			},
-			MountNsID: uint64(eventC.mntns_id),
-			Pid:       uint32(eventC.pid),
-			Comm:      C.GoString(&eventC.task[0]),
-			Dport:     uint16(C.htons(eventC.dport)),
-			Sport:     uint16(C.htons(eventC.sport)),
+			MountNsID: eventC.MntnsId,
+			Pid:       eventC.Pid,
+			Comm:      gadgets.FromCString(eventC.Task[:]),
+			Dport:     gadgets.Htons(eventC.Dport),
+			Sport:     gadgets.Htons(eventC.Sport),
 		}
 
-		if eventC.af == C.AF_INET {
+		if eventC.Af == unix.AF_INET {
 			event.IPVersion = 4
-		} else if eventC.af == C.AF_INET6 {
+			event.Saddr = netip.AddrFrom4(*(*[4]byte)(eventC.SaddrV6[:4])).String()
+			event.Daddr = netip.AddrFrom4(*(*[4]byte)(eventC.DaddrV6[:4])).String()
+		} else if eventC.Af == unix.AF_INET6 {
 			event.IPVersion = 6
+			event.Saddr = netip.AddrFrom16(eventC.SaddrV6).String()
+			event.Daddr = netip.AddrFrom16(eventC.DaddrV6).String()
 		}
 
-		if eventC._type == C.TCP_EVENT_TYPE_CONNECT {
+		if eventC.Type == tcptracerEventTypeTCP_EVENT_TYPE_CONNECT {
 			event.Operation = "connect"
-		} else if eventC._type == C.TCP_EVENT_TYPE_ACCEPT {
+		} else if eventC.Type == tcptracerEventTypeTCP_EVENT_TYPE_ACCEPT {
 			event.Operation = "accept"
-		} else if eventC._type == C.TCP_EVENT_TYPE_CLOSE {
+		} else if eventC.Type == tcptracerEventTypeTCP_EVENT_TYPE_CLOSE {
 			event.Operation = "close"
 		}
```

Comment:
I know it's not related to this PR, but can we take the opportunity to replace by a switch statement?